src/main/java/seedu/billboard/model/recurrence/RecurrenceList.java

```diff
@@ -0,0 +1,190 @@
+        }
+
+        if (!target.equals(editedRecurrence) && contains(editedRecurrence)) {
+            throw new DuplicateRecurrenceException();
+        }
+
+        internalList.set(index, editedRecurrence);
+    }
+
+    /**
+     * Removes the equivalent recurrence from the list.
+     * The record must exist in the list.
+     */
+    public void remove(Recurrence toRemove) {
+        requireNonNull(toRemove);
+        if (!internalList.remove(toRemove)) {
+            throw new RecurrenceNotFoundException();
+        }
+    }
+
+    /**
+     * Removes the equivalent recurrence at index from the list.
+     * The index must be valid.
+     */
+    public Recurrence remove(int index) {
+        requireNonNull(index);
+        try {
+            return internalList.remove(index);
+        } catch (IndexOutOfBoundsException e) {
+            throw new RecurrenceNotFoundException();
+        }
+    }
+
+    public void setRecurrences(RecurrenceList replacement) {
+        requireNonNull(replacement);
+        internalList.clear();
+        internalList.addAll(replacement.asUnmodifiableList());
+    }
+
+    /**
+     * Replaces the contents of this list with {@code Recurrence}.
+     * {@code Recurrence} must not contain duplicate Recurrence.
+     */
+    public void setRecurrences(List<Recurrence> recurrences) {
+        requireAllNonNull(recurrences);
+        if (!recurrencesAreUnique(recurrences)) {
+            throw new DuplicateRecurrenceException();
+        }
+
+        internalList.clear();
+        internalList.addAll(recurrences);
+    }
+
+    /**
+     * Returns the backing list as an unmodifiable {@code ObservableList}.
+     */
+    public List<Recurrence> asUnmodifiableList() {
+        return Collections.unmodifiableList(internalList);
+    }
+
+    @Override
+    public Iterator<Recurrence> iterator() {
+        return internalList.iterator();
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof RecurrenceList // instanceof handles nulls
+                        && internalList.equals(((RecurrenceList) other).internalList));
+    }
+
+    @Override
+    public int hashCode() {
+        return internalList.hashCode();
+    }
+
+    /**
+     * Returns true if {@code Recurrence} contains only unique Recurrence.
+     */
+    private boolean recurrencesAreUnique(List<? extends Recurrence> recurrences) {
+        for (int i = 0; i < recurrences.size() - 1; i++) {
+            for (int j = i + 1; j < recurrences.size(); j++) {
+                if (recurrences.get(i).equals(recurrences.get(j))) {
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+
+    public RecurrenceList getClone() {
+        RecurrenceList clonedList = new RecurrenceList();
+        for (Recurrence e : this.internalList) {
+            clonedList.add(e);
```

Comment:
This is not actually making deep copy of `Recurrence` objects, need to initialize a new instance of `Recurrence` with the same values